app/src/main/java/com/asfoundation/wallet/analytics/RakamEventLogger.kt

```diff
@@ -0,0 +1,55 @@
+package com.asfoundation.wallet.analytics
+
+import android.util.Log
+import cm.aptoide.analytics.AnalyticsManager
+import cm.aptoide.analytics.EventLogger
+import io.rakam.api.Rakam
+import org.json.JSONException
+import org.json.JSONObject
+
+class RakamEventLogger : EventLogger {
+
+  companion object {
+    const val TAG = "RakamEventLogger"
+  }
+
+  override fun setup() {
+
+  }
+
+  override fun log(eventName: String, data: Map<String, Any>?,
+                   action: AnalyticsManager.Action, context: String) {
+    if (data != null) {
+      Rakam.getInstance()
+          .logEvent(eventName, mapToJsonObject(data))
+    } else {
+      Rakam.getInstance()
+          .logEvent(eventName)
+    }
+    Log.d(TAG, "log() called with: "
```

Comment:
You can use strign templates here.

    Log.d(TAG,
        "log() called with: eventName = [$eventName], data = [$data], action = [$action], context = [$context]")